Telemetry Battery PCB

gumile opened the pull request.

gumile committed:
telemetry-battery-pcb: create project

gumile committed:
telemetry-battery-pcb: add LTC2944

gumile committed:
telemetry-battery-pcb: complete LTC2944 circuitry

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Vérifier/recalculer la valeur de la résistance shunt

efondere replied:
$5 m\Omega$, voir [wiki](https://github.com/Exocet-Polytechnique/exo-pcbs/wiki/PCB03-%E2%80%90-Batteries-de-t%C3%A9l%C3%A9m%C3%A9trie#2-calcul-de-la-r%C3%A9sistance-shunt)

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Est-ce que le choix des pins pour le I2C est bon? (à vérifier peut-être lors du placement des footprints sur le PCB)

efondere replied:
Tous les I2C sur le microcontrôleur ont les mêmes fonctionnalités:
<img width="730" height="286" alt="image" src="https://github.com/user-attachments/assets/ad24a672-74d7-4f52-9a0a-f290648978ce" />
(attention notre microcontrôleur n'a pas forcément autant de sorties I2C, la datasheet est générique et décrit en même temps de modèles avec plus de pins)

Il s'agit simplement de choisir les pins qui conviennent le mieux lors du routage. Vérifier sur la datasheet pour être certains qu'on utilise bien des pins I2C (au cas où le symbole KiCad n'est pas à jour).

efondere replied:
Sur le même sujet, je recommande de bouger les résistances sur le bus I2C (SCL/SDA) près des pins du stm32 ou près des _sheet pins_ dans la première page étant donné qu'elle sont communes à tous les périphériques I2C.

gumile committed:
telemetry-battery-pcb: add EEPROM (M24C02-WMN6TP)

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Est-ce que l'adresse I2C du EEPROM (M24C02-WMN6TP) est bonne? Il est possible de connecter les pins E0, E1 et E2 à VCC ou GND pour changer son adresse

efondere replied:
Oui, voir [wiki](https://github.com/Exocet-Polytechnique/exo-pcbs/wiki/PCB03-%E2%80%90-Batteries-de-t%C3%A9l%C3%A9m%C3%A9trie#adresses-i2c); pas de conflit avec le LTC2944. Garder E0 à E2 au GND.

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Que pensez-vous du EEPROM choisi? (M24C02-WMN6TP)
- Stockage: 256 bytes
- I2C

efondere replied:
Oui, c'est parfait monsieur!

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
J'ai connecté au GND la pin WC du M24C02-WMN6TP (ce qui ne bloque pas l'écriture), est-ce correct?

Référence:
<img width="854" height="203" alt="image" src="https://github.com/user-attachments/assets/a4108e3c-e5b2-453f-8f7e-ada1f29dd489" />

efondere replied:
Si on a la place pour brancher le signal sur un GPIO, je le ferais au cas où.

gumile committed:
telemetry-battery-pcb: add DC-DC Converter 24V->5V (TPS5430)

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Que pensez-vous du DC-DC Buck Converter 24V->5V (TPS5430)?
- max 3A continuous (4A peak) 
- Texas Instruments

efondere replied:
Voir [section sur le courant du wiki](https://github.com/Exocet-Polytechnique/exo-pcbs/wiki/PCB03-%E2%80%90-Batteries-de-t%C3%A9l%C3%A9m%C3%A9trie#1-courant-moyen-et-max-dans-le-r%C3%A9seau).

J'ai peur que 3 A / 4 A ça ne soit pas suffisant pour les appareils alimentés par le 5 V. J'ai trouvé le [TPSM86837](https://jlcpcb.com/partdetail/TexasInstruments-TPSM86837RCGR/C34572614), permettant un plus haut courant (et dont le circuit d'alimentation semble plus simple). Il est un peu cher et le stock est bas, donc voici aussi d'autres alternatives:
- https://jlcpcb.com/partdetail/VishayIntertech-SIC402ACD_T1GE3/C511100
- https://jlcpcb.com/partdetail/MaxLinear-XR76208ELTRF/C555356

Tu as bien fait aussi de chercher DC-DC buck converter, j'ai l'impression que les regulator c'est pour < 2 A généralement.

gumile replied:
Finalement le [TPSM86837RCGR](https://jlcpcb.com/partdetail/TexasInstruments-TPSM86837RCGR/C34572614) a été choisi et son circuit a été complété. Thread résolu 👌

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
J'ai de la misère à savoir quoi choisir comme connecteur pour fils de la batterie (ring terminal ou log terminal)

efondere replied:
N'importe quel connecteur pouvant supporter jusqu'à 10 A continu fera l'affaire. On n'est peut-être même pas obligé de prendre un ring/log terminal finalement (étant donné qu'on a plus à alimenter les contrôleurs de FC avec la batterie de télémétrie). Sinon, j'ai trouvé ça sur DigiKey: https://www.digikey.com/en/products/detail/metz-connect-usa-inc/360410/9462530 .

gumile replied:
Ce connecteur a été bel et bien utilisé

gumile committed:
telemetry-battery-pcb: corrections (battery gauge, EEPROM)

gumile committed:
telemetry-battery-pcb: assign JLCPCB# and footprints

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
ajouter le condenstaeur 
<img width="516" height="375" alt="image" src="https://github.com/user-attachments/assets/5e882e67-4b74-46a1-8cba-c437daf34e1e" />

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Tests points:
- pour ceux du DCDC buck converter, il faut mettre des pads:
<img width="474" height="338" alt="image" src="https://github.com/user-attachments/assets/49acfcb7-a74d-45db-af7f-f6662e1a2406" />

- pour les autres, copier les TP de la page 2 du schematic

gumile committed:
telemetry-battery-pcb: add MOSFETs & change DC-DC Converter 24V->5V (TPSM86837)

gumile committed:
telemetry-battery-pcb: create battery connector & rearrange schematic

gumile committed:
telemetry-battery-pcb: assign footprints (all except resistors and capacitors)

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Vérifier si certaines composantes (capacitors ou résistances) doivent supporter un voltage/ampérage plus élevé (peut-être choisir d'autres parts sur JLCPCB)

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
ajouter modèles 3D

efondere replied:
Ne sera pas fait, car trop long et pas très utile à court terme.

gumile committed:
telemetry-battery-pcb: assign footprints & JLCPCB# (resistors and capacitors)

gumile commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Changer le convertisseur 24V->5V (le TPSM86837RCGR)
[TPSM86837RCGR | Texas Instruments | DC-DC Power Modules | JLCPCB](https://jlcpcb.com/partdetail/TexasInstruments-TPSM86837RCGR/C34572614)
[Datasheet TPSM86837RCGR](https://www.ti.com/lit/gpn/TPSM86837)

- Raisons:
    - _Le TPSM86837RCGR n'est pas problématique en soit, mais il est difficile de comprendre comment il fonctionne/comment faire le circuit autour. Aussi, il est relativement cher et disponible en faible quantité sur JLCPCB. Il serait bien de trouver un composant qu'on comprend mieux._
    - On est à la limite du courant:
        - Le TPSM86837RCGR peut supporter 8A
        - Voir calcul courant: [Courant moyen et max dans le réseau](https://github.com/Exocet-Polytechnique/exo-pcbs/wiki/PCB03-%E2%80%90-Batteries-de-t%C3%A9l%C3%A9m%C3%A9trie#1-courant-moyen-et-max-dans-le-r%C3%A9seau)
- Contraintes:
    - Input 24V
    - Output 5V
    - Doit supporter le courant du circuit: [Courant moyen et max dans le réseau](https://github.com/Exocet-Polytechnique/exo-pcbs/wiki/PCB03-%E2%80%90-Batteries-de-t%C3%A9l%C3%A9m%C3%A9trie#1-courant-moyen-et-max-dans-le-r%C3%A9seau)
- Utile:
    - Site pour choisir les composants: https://jlcpcb.com/parts/
        - Attention, il y a 2 sections sur ce site correspondant à ce qu'on cherche:
            1. [JLCPCB - DC-DC Converters](https://jlcpcb.com/parts/2nd/Power_Management_(PMIC)/DC-DC_Converters_3005): Ce sont des convertisseurs plus pour de petits ampérages (pour de l'électronique à haute fréquence), ce n'est pas ce qu'on recherche
            2. [JLCPCB - Power Modules](https://jlcpcb.com/parts/2nd/Power_Modules/DC-DC_Power_Modules_2116): C'est ce qu'on recherche; des modules pour de plus grands ampérages

Nicolas3311 committed:
telemetry-battery-pcb: create XR76208 symbol

Nicolas3311 committed:
telemetry-battery-pcb:schematic

Lusser55 committed:
telemetry-battery-pcb: create TPS1HTC30-Q1 symbol

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Il y a quelque connexions qui sont superflues dans le schéma (surlignées sur la capture suivante). Vous pouvez les enlever pour l'alléger.

<img width="1526" height="648" alt="image" src="https://github.com/user-attachments/assets/f43437f6-434a-4d01-9209-ad238ccbcdd0" />

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Étant donné que moins de courant passe par le 5V finalement, on peut baisser RLIM (R12 + R24 sur le schéma) à 4.3 kOhm (Iocp = 8.63A), ce qui évitera d'avoir besoin de deux résistances.

efondere replied:
Mettre à jour le numéro JLCPCB (en ce moment c'est encore 4.3k). Voici une résistance ok: https://jlcpcb.com/partdetail/18355-0805W8F4301T5E/C17667

Par contre, je me relis et je me demande si je voulais pas plutôt dire 4.7k (refaire les calculs et voir quelle résistance donne Iocp = 8.63A entre les deux). Si on garde la 4.7k, je crois que la pièce JLCPCB actuelle est ok.

Aussi, il faudrait idéalement garder la référence comme RX plutôt que RLIM1. On peut ensuite ajouter RLim dans la description du symbole.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Pour éviter d'utiliser des résistances extended, on peut approximer la résistance R1 pour la programmation de Vout (R12 + R25 sur le schéma) par 15k // 680k (j'ai utilisé la calculatrice de KiCad pour ça). 

- 15k: https://jlcpcb.com/partdetail/23536-0603WAF1502T5E/C22809
- 680k: https://jlcpcb.com/partdetail/18485-0805W8F6803T5E/C17797 (c'est _extended_, mais avec un pouce, @gumile tu me confirmeras si c'est ok)

gumile replied:
Je pense que tu parle plutôt de: R13 + R25 = 14k + 600 = 14600 Ohm
Dans ce cas le calcul est bon: 15k // 680k = 14676 Ohm
Les modèles de résistances que tu as donnés sont bons (basic et extended avec un pouce c'est ok)

efondere replied:
Vérifier les footprints + numéros JLCPCB (je crois que ça n'a pas été mis à jour)

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Pour Cout (C24 sur le schéma), on peut utiliser 2 * 47 uF, ce qui donne Flc = 9.04kHz <= 11kHz. Ce sont des valeurs plus standard (les mêmes que dans leur schéma, juste moins).

Condensateur de 47uF en céramique basic sur JLCPCB (condition pour utiliser Cff, donc j'imagine que c'est recommandé -- en plus les électrolytiques sont pas basic): https://jlcpcb.com/partdetail/17464-CL21A476MQYNNNE/C16780

Avec cette valeur et R1 = 15k // 680k, j'obtiens Cff ~= 171 pF. En faisant quelques recherches, je suis tombé sur ça à propos du condensateur feedforward, je vous le partage pour référence:
> [...] high frequency disturbances on the output will have a lower impedance path to the FB pin with the addition of the FF capacitor. [...] So by adding the FF capacitor it's possible to improve the transient performance of the converter.
(source: https://electronics.stackexchange.com/questions/634289/what-is-the-function-of-the-feedforward-capacitor-on-the-lm2596-adj)

Il y a des condensateurs 150pF en basic qu'on peut utiliser pour Cff: https://jlcpcb.com/partdetail/1946-0603B151K500NT/C1594

efondere replied:
Pour les deux Cout: le numéro JLCPCB + footprint doit être mis à jour.

Pour Cff: :+1:

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Pour R4 et R3 (R23 et R26 sur le schéma): elles permettent de choisir entre le mode DCM/CCM ou CCM seulement du convertisseur. DCM/CCM permet d'avoir une meilleure efficacité à bas load. Étant donné qu'il risque d'y avoir des moments où ça sera le cas (ex. pour mesurer la charge de la batterie), je pense qu'on devrait utiliser ce mode-là. Vous pouvez vous fier à cette figure de la page 13 de la datasheet pour le circuit:
<img width="560" height="479" alt="image" src="https://github.com/user-attachments/assets/2566a6de-d1ef-4b55-a5a4-de7a3ef38d23" />

De ce que je comprend, il faudrait utiliser une diode Zener avec une polarité inverse d'environ 3.3V (Vin représente +24V dans notre cas).

Les autres schémas utilisent des diviseurs de tension, mais ils dépendent de la tension d'alimentation, ce qui peut poser des problèmes si la batterie n'est pas assez chargée (et donc Vin est < 24V) et que la tension à la pin EN/MODE baisse en dessous de 3.1V.

efondere replied:
Il faut trouver une diode zener avec les caractéristiques requises et ajouter son numéro JLCPCB (ajouter le lien dans cette discussion au besoin). Ensuite, il faut choisir le footprint selon la description sur JLCPCB. La datasheet recommande la diode MMSZ4685T1G, mais je crois qu'elle n'est pas basic sur JLCPCB. Vous pouvez faire une petite recherche pour des équivalents basic, sinon on prendra celle-là.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Quelques composants / circuits à ajouter mais qui ne sont pas dans la datasheet:
- Test points connectés aux pins (voir page avec le STM32 pour exemples de test points):
  - PGOOD
  - EN/MODE
  - VCC
  - FB
  - SW
- Circuit avec MOSFET pour allumer/éteindre une diode selon l'état de PGOOD:
  - PGOOD est open drain: sa sortie est 0 quand il y a un problème (voir p. 6)
  - La résistance R10 sert de pull-up à VCC (environ 5V, voir p. 3)
  - On veut allumer la diode dans le DC-DC fonctionne correctement
  - Il faut une résistance d'environ 250 à 750 ohm en série avec la DEL (dépend de la couleur, voir la datasheet de la DEL utilisée)

Bien sûr on peut ajouter plus de test points si on veut, mais je pense qu'avec ceux que j'ai mentionnés, on sera en mesure de débugger le DC-DC en cas de problème.

efondere replied:
On en a déjà parlé, mais les TP pour VCC, FB et SW ne sont pas connectés. Allez voir aussi les paramètres des symboles pour les TPs, vous pouvez remplacer le Field "Net Label" par le bon (en ce moment ils sont à CAN_RX)

Pour le MOSFET et la DEL:n
- il manque la pièce JLCPCB du MOSET et de la DEL ainsi que quelques autres pièces
- Je ne vois pas l'utilité du condensateur C30: peut-être que j'ai juste oublié une discussion qu'on a eu?
- Je vous conseile de réutiliser le label PGOOD pour éviter d'avoir un fil (celui qui est en haut de R21) qui en traverse plusieurs autres.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
Tel que mentionné en réunion, il faudra faire un footprint pour le DC-DC où plusieurs pads vont partager le même numéro.

efondere reviewed (commented):
Voici quelques notes plus génériques, mais qui ne nécessitent pas de changement:
- Cin, CVcc, Cin1: [condensateurs de découplage](https://en.wikipedia.org/wiki/Decoupling_capacitor); on les laisse tel que décrits dans leurs exemples
- Cbst: [condensateur de bootstrap](https://en.wikipedia.org/wiki/Bootstrapping_(electronics)), on le laisse aussi tel que décrit
- Je n'ai pas vérifié tous les composants sur JLCPCB. Si ce n'est pas déjà fait:
  - Vérifier basic vs extended
  - Vérifier la tension maximale des condensateurs
  - Vérifier la polarité si nécessaire
- Je me suis arrêté au commit 90dd150c22dfbb2b0e6ea37b0b7b5111147b8b07 (telemetry-battery-pcb: schematic) pour cette première _review_

Lusser55 committed:
telemetry-battery-pcb: create XR76208 Footprint

Lusser55 committed:
telemetry-battery-pcb: led circuit added

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Je crois qu'il y a plusieurs pièces qui ont été changées ce qui veut dire que leur numéro JLCPCB et le footprint ne sont plus à jour. Vous pouvez utiliser la fenêtre "Symbol Fields Table" pour vous aider: 

<img width="1576" height="910" alt="Image" src="https://github.com/user-attachments/assets/6f76c968-1901-48f6-86bd-7c72d913f230" />

Je vous conseille de choisir "Current sheet only" dans "Scope" en bas pour ne pas accidentellement modifier les numéros des symboles la page du layout commun.

Vous pourrez aussi en profier pour ajouter les numéros JLCPCB aux pièces qui n'en ont pas (à l'exception des pièces avec des X rouges dessus, car on va les assembler nous-mêmes)

efondere commented on _library/footprint/Exocet.pretty/MaxLinear_XR76208ELTR-F_QFN-30(5x5).kicad_mod:
Ajouter un petit cercle à côté de la pin 1 (c'est standard pour indiquer l'orientation d'une chip). Pour se faire:
1. Choisir la layer "F.Silkscreen" à droite
2. Utiliser l'outil cercle et créer un cercle
3. Double-cliquer sur le cercle pour ouvrir les paramètres: changer le rayon pour quelque chose de raisonnable (ça risque d'être difficile avec la grille par défaut autrement) et sélectionner "Filled shape"
4. Placer le cercle à côté de la pin 1 (juste en-dessous c'est good par exemple)

efondere commented on _library/footprint/Exocet.pretty/MaxLinear_XR76208ELTR-F_QFN-30(5x5).kicad_mod:
Le petit décalage au niveau de la pin 11 est un peu étrange, mais après avoir mesuré pas mal, j'ai l'impression que c'est plus un problème de la datasheet. Dans tous les, cas, je crois que ça vaut pas la peine de s'en soucier.

efondere commented on _library/footprint/Exocet.pretty/MaxLinear_XR76208ELTR-F_QFN-30(5x5).kicad_mod:
Il existe déjà une librairie appelée "Exocet" pour les footprints. Vous pourrez copier celui que vous avez créé et le mettre à l'intérieur, comme ça on peut tout grouper ensemble. Pour la trouver, vous pouvez chercher "Exocet" dans Libraries dans l'éditeur de footprints.

Lusser55 committed:
telemetry-battery-pcb: modify zener jlcpcb code

Nicolas3311 committed:
telemetry-battery-pcb: updated JLC parts and footprints

Lusser55 commented:
[TPS1HTC30-Q1(calculs).pdf](https://github.com/user-attachments/files/24899364/TPS1HTC30-Q1.calculs.pdf)
-erreur sur le rapport de puissance: c'est plutot: Pout/Pin.

Lusser55 commented:
<img width="1034" height="500" alt="image" src="https://github.com/user-attachments/assets/eebcf6e2-1aa6-4d5c-8be0-9b273ff08adc" />
- Calcul pour la led controlé par la pin Pgood du DC-DC

Lusser55 commented:
https://www.ti.com/lit/an/slvae30e/slvae30e.pdf?ts=1769520046976
-Documentation sur la programmation des smart switch(TPS), pour allimenter divers charges(capacitif, inductifs etc.)

Lusser55 committed:
telemetry-battery-pcb: connections for  TPS1HTC30-Q1

Lusser55 committed:
telemetry-battery-pcb: schematic connections complete

Lusser55 committed:
telemetry_battery: modify footprint tps1htc30

Lusser55 committed:
telemetry battery: move footprint XR76208 to Exocet Library

Lusser55 committed:
telemetry-battery: start of routing

Lusser55 committed:
telemetry-battery: Placements of parts

Lusser55 committed:
telemetry-battery: modify footprint P-channel-Mosfet

Lusser55 committed:
telemetry-battery : Routing of PCB

Nicolas3311 committed:
Telemetry:Presque finis Routage

efondere committed:
telemetry-battery-pcb: review schematic and pcb

Changes:
- Reduce the value of fuse F2 to 10A (no impact on footprint or JLCPCB
  number)
- Group all test points together (and relevant changes to PCB)
  - +24V test point
  - I2C test points (SDA and SCL)
- Change the 24V-network used as input for the 5V DC-DC (and relevant
  changes to PCB). The STM32 on this PCB needs to be powered on (using
  the 5V-network) in ordrer to enable the 24V-net output for other PCBs.
  Therefore, the DC-DC converter needs to be supplied directly from the
  battery, as described in the TPS1HTC30 datasheet.
- Change the position of the +5V power label (and relevant changes to
  PCB). The reasoning is similar to the previous point. The 5V_NET is
  controlled by the STM32 itself, which therefore needs to be powered on
  separately.
- Change the 24V-network used for the PGOOD LED (results from changing
  the 24V-nework input to the DC-DC converter). PCB was also updated to
  reflect these changes.
- Connect the TPS1HTC30 I/O to STM32. There was an incompatibility with
  the labels used which meant KiCad didn't properly connect the nets.
  Now we have local labels in the root page, connected to hierarchical
  labels from the STM32-page. Also swapped the pin used by SNS to PA0,
  since it needs to be connected to an ADC (PA0 is input 1 of ADC 1).
  PCB was NOT updated accordingly.
- Reorganize and cleaned some labels and text in the schematic
  (alignment, etc.) for clarity.
- Change the style of the DC-DC converter's pin 1 dot in the footprint.
  Also added traces between the inernally connected pins to remove the
  DRC warnings.
- Update PCB metadata (name, code, authors, reviewers, etc.). Also
  updated the text in the Silkscreen for the name of the pcb to match
  what there is in the template.
- Simplify routing in the PCB and removed redundant traces

efondere commented on telemetry-battery-pcb/stm32_base.kicad_sch:
Pour la diode D6 (au niveau du `GND_L` du TPS1HTC30): TI recommande une diode avec une tension inverse de 600V et un courant de 0.2A. La diode actuelle respecte la caractéristique pour le courant, mais pas pour la tension (max 100V). Idéalement, on devrait la remplacer pour respecter les recommandation étant donné qu'on va avoir plusieurs bobines dans les *coils* de nos contacteurs du système de haute puissance.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Pour la diode D6 (au niveau du `GND_L` du TPS1HTC30): TI recommande une diode avec une tension inverse de 600V et un courant de 0.2A. La diode actuelle respecte la caractéristique pour le courant, mais pas pour la tension (max 100V). Idéalement, on devrait la remplacer pour respecter les recommandation étant donné qu'on va avoir plusieurs bobines dans les *coils* de nos contacteurs du système de haute puissance.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Tel que discuté, il faut mettre à jour les valeurs des composants pour le convertisseur DC-DC étant donné qu'il y aura moins de courant que prévu. Voir le google sheets partagé sur Discord pour info. Dites-moi quand c'est fait et je réviserai une dernière fois cette partie du schéma.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Plusieurs composants sont extended (diode D3, résistances R18 et R19, mais il y en a peut-être d'autres),  mais on probablement des équivalents assez proches en basic. Il faudrait les changer (et peut-être leur valeur aussi) pour utiliser des basic à la place.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
Ajouter le nom du net des test points pour les repérer plus facilement. Aussi, ça peut être mieux de les éloigner des composants, car ils sont un peu dans le chemin pour le routage (c'est pas obligatoire, mais ça peut être plus facile pour la suite)

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
Il manque plusieurs connections entre le STM32 et les IO du TPS, car les labels n'étaient pas bien référencés dans le schéma.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
Pour le routage du TPS, suivez le plus possible la datasheet à la page 34: https://www.ti.com/lit/ds/symlink/tps1htc30-q1.pdf?ts=1772309406015 qui vous indique où placer vos gnd etc dans le cas d'un ground networok comme on a. Vous pouvez faire les "grosses traces" à l'aide de plusieurs planes différents

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
Il faut agrandir la trace entre la pin 2 de la bobine L1 et la source du PMOS Q2

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
À la fin, on validera que toutes les traces et composants peuvent tolérer le courant nécessaire sur les chemins où il y aura beaucoup de courant.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
Il faudra valider les dimensions du PCB et des trous de montage et éviter d'avoir des dimensions fractionnelles.

efondere reviewed (changes requested):
J'ai laissé plusieurs commentaires dans les fichiers du PCB et du schéma. J'ai aussi fait un push avec les changements suivants (voir la description de mon commit pour plus de détails):

- Changer la valeur de F2
- Grouper les test points ensemble
- Changer le réseau +24V en entrée du convertisseur DC-DC et de la DEL PGOOD *
- Changer la position du label +5V *
- Connecter les entrées / sorties du TPSHTC30 au STM32. Le PCB n'est pas à jour à ce niveau, il reste quelques traces à connecter! *
- Réorganiser et déplacer des labels/champs de texte dans le schéma
- Changer le style du point à la pin 1 pour le footprint du convertisseur DC-DC
- Ajouter des traces entre les pin connectées à l'interne du DC-DC *
- Mettre à jour les méta-données du PCB (nom, auteurs, etc.) et changer le texte du Silkscreen pour ressembler au template
- Simplifier le routage et enlever des traces redondantes *

*: On s'en reparle par Discord ou en vrai!

Il me reste encore à valider les numéros JLCPCB, etc. mais on pourra faire ça à la fin.

Lusser55 committed:
Telemetry-battery: Routing of PCB
changes:
- add new diode model for tps1htc30-Q1
- Finish routing of tpshtc30-Q1 / DRC no error

Lusser55 committed:
Telemetry-battery: Add thermal vias for Tps1htc30-Q1

Lusser55 committed:
Telemetry-battery: Change Rlim value to 1,5k ohm based on new data

efondere committed:
telemetry-battery-pcb: add temperature sensor

Added a JST PH connector for a DS18B20 1-wire temperature sensor. Also
included the DS2484Q chip to convert 1-wire signals to i2c. All new
devices/chips powered by the +5V_net and will therefore not be
functional nor consume power before all of the network is powered on.

The i2c addresses were also added close to each chip using that
protocol to make sure there will be no conflict.

efondere committed:
telemetry-battery-pcb: move certains traces and vias

Swapped nFAULT and LATCH signals to avoid vias on one of them. Removed
90-degree corners on some traces. Move fuse for shorter traces.

efondere committed:
telemetry-battery-pcb: add info to silkscreen

- Added net names to test points
- Added battery polarity info
- Added fuse current
- Added PCB version
- Added connector names and reset button label

efondere committed:
telemetry-battery-pcb: round pcb dimensions

- Moved edge cuts and planes
- Moved mounting holes

efondere committed:
telemetry-battery-pcb: add GND_L plane

For better thermal relief.

efondere committed:
telemetry-battery-pcb: set DS18B20 connector as DNP

efondere committed:
telemetry-battery-pcb: remove layers from DC-DC symbol

efondere committed:
telemetry-battery-pcb: enlarge +5V net after coil

efondere committed:
telemetry-battery-pcb: update JLCPCB component numbers

efondere commented:
Update pour le capteur de température: voir #31

efondere committed:
telemetry-battery-pcb: change DS2484 symbol

Using the DS2484R instead of DS2484Q as it is cheaper on JLCPCB.

Lusser55 committed:
telemetry-battery-pcb: change value of resistance
- R27
- C29
- C27/C28
and others

Lusser55 committed:
telemetry-battery-pcb: increase vias to 0.3mm  of the tps1htc30-q1

Lusser55 committed:
telemetry-battery-pcb: Review of pcb components and values

Lusser55 committed:
telemetry-battery-pcb : Update PCB

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Le field JLCPCB de certains composant a un espace devant, il faudrait l'enlever pour pouvoir les grouper dans le BOM:
- D4
- C30

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
C25: Changer la valeur à 100n plutôt que 0.1u  pour le grouper avec les autres de la même valeur

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Mettre le fuse holder F2 en "Do not populate"

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_pcb:
Il y a un warning de DRC, car il y a un silkscreen (R27) qui est trop près d'une pin

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Remplacer les résistances qui contrôle la limite de courant par des résistances through-hole qu'on pourra remplacer nous-mêmes.

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
- Changer C19 pour un basic: C1622 (il faudra changer le footprint pour 0603).
- Changer R11 pour un extended + pouce: C23172 (changer le footprint pour 0603).
- C31et C33: changer le footprint pour 0603 (pour matcher la pièce JLCPCB déjà choisie)
- R22: Vérifier la valeur, sur le schéma c'est 2k, mais la pièce JLCPCB c'est 4.3k; allez voir vos calculs pour garder la bonne et changer celle qu'il faut. Aussi éviter de prendre une résistance extended si possible
- R30: Changer le footprint pour 0603 (pour matcher la pièce JLCPCB déjà choisie)

efondere commented on telemetry-battery-pcb/telemetry-battery-pcb.kicad_sch:
Remplacer le numéro JLCPCB du DC-DC par C1355299. C'est la même puce, mais le packaging change (elle est un peu plus cher malheureusement, mais l'autre n'est plus en stock)

Nicolas3311 committed:
telemetry-battery-pcb: components numbers JLCPCB and footprints change

Nicolas3311 committed:
telemetry-battery-pcb: Changes on footprints Rlim and R30

efondere committed:
telemetry-battery-pcb: add fixes for template

- Added an extra footprint for a fuse
- Increased LED resistor value
- Rounded board corners
- Changed temperature sensor connector for horizontal version (not from
  template)

efondere committed:
telemetry-battery-pcb: add combined fuseholder footprint

This gets rid of all remaining DRC warnings and makes the schematic a
little clearer.

efondere committed:
telemetry-battery-pcb: add 5V-network switch ic

This switch replaces the previous MOSFET circuit, as we weren't certain
that circuit would work. Also chaned a few capacitor and resistor
footprints to reduce amount of different components that we use.

efondere committed:
telemetry-battery-pcb: change SNS to 5V-tolerant pin

efondere committed:
telemetry-battery-pcb: reorganize schematic

Makes it easier to follow the power distribution tree, although not all
power managements ICs are on the front page.

efondere committed:
telemetry-battery-pcb: fix ERC errors/warnings

efondere committed:
telemetry-battery-pcb: review 24V network switch

Also reverted to the preivous chip we wanted to use that was no longer
available, but is now.

efondere committed:
telemetry-battery-pcb: fix i2c pins on some ICs

The SDA and SCL pins were inverted on the EEPROM and one-wire to i2c
converter.

efondere committed:
telemetry-battery-pcb: change capacitor part numbers

Some capacitors did not support the required voltage. Also, some had the
incorrect footprint and were updated.

Also added some test points.

efondere committed:
telemetry-battery-pcb: reduce crossings in routing

The power input terminals were moved to the bottom left to reduce the
amount of crossings of high-current traces. Tests points were also moved
slightly as the added test points did not fit well. Through-hole
resistors were moved to the top to reduce mounting space required
underneath. Rectangular zones were added to Silkscreen to visualize
different parts of the PCB.

efondere committed:
telemetry-battery-pcb: update LTC2944 JLCPCB number

efondere committed:
telemetry-battery-pcb: update data on board layout

efondere committed:
telemetry-battery-pcb: change regulator part

PMB: increased the max current of the regulator after it stopped
working.
